safe/common/parameters/default_select_parameter_widget.py

```diff
@@ -151,3 +152,29 @@ def toggle_custom_value(self):
             self.custom_value.setDisabled(False)
         else:
             self.custom_value.setDisabled(True)
+
+    def toggle_input(self):
+        """Change behaviour of radio button based on input."""
+        # If input is a field, then disable all radio button.
+        if self.input.currentText() != 'No Field':
```

Comment:
We have data, but the content is the same as the text.
```
            self.input.addItem(opt)
            self.input.setItemData(index, opt, Qt.UserRole)
```
I think it's fine to use that @myarjunar.